CRM/CiviAwards/Form/AwardReview.php

```diff
@@ -368,28 +368,88 @@ private function getCaseTypeName() {
   }
 
   /**
-   * Gets the tags for a case.
+   * Load the tags for a case.
    *
    * @return array|string
    *   Case tags.
    */
-  private function getCaseTags() {
+  private function loadCaseTags() {
     $result = civicrm_api3('EntityTag', 'get', [
       'sequential' => 1,
       'entity_table' => 'civicrm_case',
       'api.Tag.getsingle' => ['id' => "\$value.tag_id"],
     ]);
 
     if ($result['count'] == 0) {
-      return '';
+      return [];
     }
 
-    $caseTags = '';
+    $caseTags = [];
     foreach ($result['values'] as $caseTag) {
-      $caseTags .= $caseTag['api.Tag.getsingle']['name'] . ', ';
+      $caseTag = [
+        'id'               => $caseTag['api.Tag.getsingle']['id'],
+        'parent_id'        => $caseTag['api.Tag.getsingle']['parent_id'],
+        'name'             => $caseTag['api.Tag.getsingle']['name'],
+        'background_color' => !empty($caseTag['api.Tag.getsingle']['color']) ? $caseTag['api.Tag.getsingle']['color'] : '#ffffff',
+      ];
+      $caseTag['color'] = $this->getTextColor($caseTag['background_color']);
+      $caseTags[] = $caseTag;
     }
 
-    return rtrim($caseTags, "', '");
+    return $caseTags;
+  }
+
+  /**
+   * Get Case tags list.
+   *
+   * @param string $format
+   *   (Optional) Output format, defaults to plain text.
+   *
+   * @return string
+   *   Case tags list.
+   */
+  private function getCaseTags($format = 'plain') {
+    switch ($format) {
+      case 'badge':
+        $res = [];
+        foreach ($this->caseTags as $caseTag) {
+          $res[] = '<span class="crm-tag-item" style="background-color: ' . $caseTag['background_color'] . '; color: ' . $caseTag['color'] . '">' . $caseTag['name'] . '</span>';
+        }
+
+        $res = implode(' ', $res);
+        break;
+
+      default:
+        $res = implode(', ', array_column($this->caseTags, 'name'));
```

Comment:
Do we really need the default/plain format? Can't see it being called anywhere?
If not we can remove the switch and just make the function return  by default what the badge format returns.

Reply:
There is a line 194 which uses 'plain' format:
```
    $this->assign('caseTags', $this->getCaseTags());
```

Not sure where this is used and could that be changed to use 'badge' format.
So we can make 'badge' format by default if you want, just want to confirm that we should.

Reply:
Line 194 sets the `caseTags` to be used in the template but on checking `https://github.com/compucorp/uk.co.compucorp.civiawards/blob/0bce88d517410995dc5bbe1f450e28dd730c451d/templates/CRM/CiviAwards/Form/AwardReview.tpl` I can't seem to find that used anywhere. 

So it's safe to refactor the function.

Reply:
What should we do with line 194?
1. Remove it.
2. Use 'badge' format (which would be default after refactoring).
3. Use 'plain' format still.

Reply:
Remove line 194, refactor function to only return badge style, no need for function to accept any parameter since there are no other format.
Do we still need plain? It's not used anywhere